### PR TITLE
feat(encryption): support for kms rbac

### PR DESCRIPTION
**Pull request by gannons**
# Pull Request Template

## Description

Adding support for KMS's RBAC feature. RBAC allows keys to be retrieved on behalf of other users. But only if

- the requestor has the spark.kms_orgagent role, which is assigned to compliance officers
- the compliance officer is making the request on behalf of another user within his/her organisation

This feature will be used by eDiscovery v2.

Fixes # (issue)
NA

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update (I assume?)

## Test Coverage

The provided Integration tests cover the use cases. Additional testing has been carried out using, the yet to be merged, eDiscovery module.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by gannons** [comment]
The value of **OnBehalfOf** changes between calls. We use it to retrieve a key that can decrypt an activity's content. When doing this we pass in the activity's actor id.  Meaning the id changes depending on who created the activity.

In typical usage you likely won't set anything for **onBehalfOf**. Users will retrieve keys they've access to themselves.

The only people who'll retrieve keys _on behalf of_ someone else will be compliance officers. These users have a role that allows them to retrieve a key for any other user within their organisation. With it they can decrypt space content and use that content to compile a report for legal requests, etc.

File: packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js [hunk 1]
```diff
@@ -34,13 +34,13 @@ const Encryption = SparkPlugin.extend({
       });
   },
 
-  decryptScr(key, cipherScr) {
-    return this.getKey(key)
+  decryptScr(key, cipherScr, onBehalfOf) {
```
Comment (mattnorris) [inline packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js]: What is `onBehalfOf`? Please document the type of parameter it is and an example.

Comment (gannons) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js]: Documentation added

Comment (gannons) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/encryption.js]: @mattnorris gentle reminder

File: packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js [hunk 2]
```diff
@@ -401,7 +406,7 @@ const KMS = SparkPlugin.extend({
       .catch((reason) => {
         if (process.env.NODE_ENV === 'test' && (reason.status === 403 || reason.statusCode === 403) && reason.message.match(/Failed to resolve authorization token in KmsMessage request for user/)) {
           this.logger.warn('kms: rerequested key due to test-mode kms auth failure');
-          return this.request(payload);
+          return this.request(payload, undefined, onBehalfOf);
```
Comment (mattnorris) [inline packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js]: This is a bit awkward. It's probably best to create the second argument as an options object with `timeout` and `onBehalfOf`.

Comment (gannons) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js]: Good call @mattnorris - I've switched both the KMS and Encryption APIs over to use an option object.

Comment (gannons) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js]: Any other change you'd like to see @mattnorris?

Comment (mattnorris) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js]: Looks good, @gannons

Comment (gannons) [reply packages/node_modules/@ciscospark/internal-plugin-encryption/src/kms.js]: Perfect @mattnorris - how does a merge work on the sdk? Is it scheduled or do people commit adhoc?

**Pull request merged**